From: Nathan Pratta Teodosio <nathan.teodosio@canonical.com>
Date: Mon, 26 Sep 2022 14:41:21 -0300
Subject: Allow {get,set}_mempolicy syscalls

They cause builds to fail on Debian with

"Disallowed syscall "get_mempolicy" caught in sandbox"

Closes: https://gitlab.gnome.org/GNOME/tracker-miners/-/issues/234
(cherry picked from commit e298a2f0e28258591e6d9178678f43b34e731c33)
---
 src/libtracker-miners-common/tracker-seccomp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/libtracker-miners-common/tracker-seccomp.c b/src/libtracker-miners-common/tracker-seccomp.c
index 88fbbe1..2f9cb81 100644
--- a/src/libtracker-miners-common/tracker-seccomp.c
+++ b/src/libtracker-miners-common/tracker-seccomp.c
@@ -102,6 +102,8 @@ tracker_seccomp_init (void)
 
 	/* Memory management */
 	ALLOW_RULE (brk);
+	ALLOW_RULE (get_mempolicy);
+	ALLOW_RULE (set_mempolicy);
 	ALLOW_RULE (mmap);
 	ALLOW_RULE (mmap2);
 	ALLOW_RULE (munmap);
